### ISSUE TITLE
Mechanic Changes on copy effects cause several bugs (Chameleos, Sideshow Spelleater)

### ISSUE THREAD

**Issue by Kybxd** [HearthSim/hs-bugs#1007]
Video: https://www.bilibili.com/video/av28709368

(Starting from 3:40) Chameleos transforms into enemy Molten Blade, then he continues to transform as an enemy Molten Blade, i.e. he transforms into a random weapon at the start of every opponent's turn.

(Starting from 4:25) Chameleos transforms into enemy Chameleos, then he transforms into a random copy of your hand at the start of every opponent's turn.

(Note that before this patch if Chameleos becomes Zerus or another shifting card, it continues to transform in Chameleos' way at the start of your next turn.)

(Starting from 5:30) If enemy hero power is turned off when you play Sideshow Spelleater, you will copy a turned-off hero power and cannot use it immediately.

**Comment by Kybxd** [comment]
> Sideshow Spelleater will no longer sometimes cause a player’s Hero Power to remain inactive.

**Comment by troggnostupidhs** [comment]
Also Floop

https://www.reddit.com/r/hearthstone/comments/9m4en6/if_the_chameleos_in_your_hand_shifts_to_floop_and/

> If the Chameleos in your hand shifts to Floop, and you play a card, it will remain as a 4 cost version of the last card you played. I guess this means it’s working as advertised but I thought the changing Floop would revert back to Chameleos at the end of your turn.

![image](https://user-images.githubusercontent.com/22508641/46582191-3f4bff00-ca11-11e8-821d-91dfd5f767c7.png)

**Comment by troggnostupidhs** [comment]
https://www.reddit.com/r/hearthstone/comments/9r5pe3/i_found_a_new_bug_interaction_with_chameleos/

**Comment by Kybxd** [comment]
https://www.bilibili.com/video/av48408255

The interaction between Chameleos & other shifting card is changed. In this video Chameleos transforms into opponent's Shifting Scroll. Then at the start of your next turn, it transforms into a random card of opponent's hand if you're Dominant Player, or transforms into a random Mage spell if opponent is Dominant Player. The reason might be "Shifting cards keep no buff when they shift". So whichever buff triggers first will clear the second buff.

Note that if Shifting Scroll wins, the card will lose the blue animation which indicates this card continues shifting. (1:00 to 2:05)

**Comment by Jetz72** [comment]
Would that suggest that the enchantment that allows them to shift is still owned by the opposite player, even though it's copied to a friendly card?

**Comment by Kybxd** [comment]
Right, and this rule also applies to Faceless Manipulator. For example, opponent cast Power Word Glory on his Tuskboar and you Faceless it. Your boar's attack healing or damaging enemy hero is still decided by whether opponent controls Auchenai or not.
